g2p_registry_base/models/registrant.py

```diff
@@ -55,3 +60,115 @@ def enable_registrant(self):
+            batch_cnt = (
+                self.env["ir.config_parameter"]
+                .sudo()
+                .get_param("g2p_registry.batch_registrants_count_job_queue")
+            )
+            try:
+                batch_cnt = int(batch_cnt)
+            except Exception:
+                batch_cnt = 2000
+
+            # Todo: Divide recordset (self) to batches by batch_cnt
+            self.with_delay()._process_batch_update_compute_fields(
+                self, field_name, kinds, indicators
+            )
+
+            # Send message to admins via odoobot
+            message = _(
+                "The processing of the calculated field: %(field)s with %(cnt)s records "
+                + "was put on queue. You will be notified once the process is completed."
+            ) % {"field": field_name, "cnt": tot_rec}
+            self._send_message_admins(message)
+
+    def _send_message_admins(self, message):
+        """Adopt OdooBot Send a message to group g2p_registry_base.group_g2p_admin users
+
+        :param user: 'res.users'  object
+        :param message: str,  The message content
+        """
+        # Obtain the OdooBot ID
+        odoobot_id = self.env["ir.model.data"]._xmlid_to_res_id("base.partner_root")
+
+        # Obtain OdooBot Chat channels of g2p_registry_base.group_g2p_admin users
+        admin_group_id = self.env["ir.model.data"]._xmlid_to_res_id(
+            "g2p_registry_base.group_g2p_admin"
+        )
+        admin_group = (
+            self.env["res.groups"].sudo().search([("id", "=", admin_group_id)])
+        )
+        channel_users = admin_group.mapped("users")
+
+        for user in channel_users:
+            channel = (
+                self.env["mail.channel"]
+                .sudo()
+                .search(
+                    [
+                        ("channel_type", "=", "chat"),
+                        ("channel_partner_ids", "in", [user.partner_id.id]),
+                        ("channel_partner_ids", "in", [odoobot_id]),
+                    ],
+                    limit=1,
+                )
+            )
+
+            #  If it does not exist, initialize the chat channel
+            if not channel:
+                user.odoobot_state = "not_initialized"
+                channel = self.env["mail.channel"].with_user(user).init_odoobot()
+            #  Send a message
+            channel.sudo().message_post(
+                body=message,
+                author_id=odoobot_id,
+                message_type="comment",
+                subtype_xmlid="mail.mt_comment",
+            )
+
+    def _process_batch_update_compute_fields(
+        self, records, field_name, kinds, indicators
+    ):
+        for record in records:
+            if record["is_group"]:
+                record[field_name] = record.count_individuals(
+                    kinds=kinds, indicators=indicators
+                )
+            else:
+                record[field_name] = None
```

Comment:
We should avoid having to loop through records. We should work by batch.

Reply:
- The function: _compute_count_and_set call to count_individuals will generate and execute a single query to generate values for all record's compute fields that needs to be updated.
- The function: query_members_aggregate generates a list of dict with this format:
> [
>    {'id': 1, 'members_cnt': 3},
>    {'id': 5, 'members_cnt': 6}
> ]

The only loop is performed in the updating of the compute field values.